src/Negotiation/AbstractNegotiator.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Negotiation;
+
+abstract class AbstractNegotiator
+{
+    /**
+     * @param string $header     A string containing an `Accept|Accept-*` header.
+     * @param array  $priorities A set of server priorities.
+     *
+     * @return BaseAccept best matching type
+     */
+    public function getBest($header, array $priorities)
+    {
+        if (!$priorities) {
+            throw new \Exception('no priorities given'); 
```

Comment:
Any error from parsing the headers should be named exceptions to be catched easily (and if you have several different failure ways, there should be several named exception to distinguish them and a common marker interface or base class to catch them all easily).
For bad arguments, there is 2 choices IMO: either using named exceptions as well or using ``\InvalidArgumentException``. ``\InvalidArgumentException`` may be enough for such case though